src/adapters/MeetingsJSONAdapter.js

```diff
@@ -201,6 +227,88 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
     return stream;
   }
 
+  /**
+   * Toggles muting the local audio media stream track.
+   * Used by "mute-audio" meeting control.
+   *
+   * @param {string} ID  ID of the meeting for which to mute local audio.
+   * @memberof MeetingsJSONAdapter
+   * @private
+   */
+  async toggleMuteAudio(ID) {
+    if (this.datasource[ID]) {
+      const meeting = this.datasource[ID];
+
+      if (meeting.localAudio) {
+        meeting.localAudio = null;
+      } else {
+        meeting.localAudio = await this.getStream({video: false, audio: true});
+      }
+
+      document.dispatchEvent(new CustomEvent(MUTE_AUDIO_CONTROL, {detail: meeting}));
+    }
+  }
+
+  /**
+   * Toggles muting the local audio media stream track.
+   * Used by "mute-video" meeting control.
+   *
+   * @param {string} ID  ID of the meeting for which to mute local audio.
+   * @memberof MeetingsJSONAdapter
+   * @private
+   */
+  async toggleMuteVideo(ID) {
+    if (this.datasource[ID]) {
+      const meeting = this.datasource[ID];
```

Comment:
This is why the rending bug was showing up. If you look at the previous implementation, a new object was always created and returned. Here we were sending back the same object

Reply:
right. the static data needs to be updated though.

Reply:
We should a copy of the meeting before modifying it, to treat it as "immutable"
```suggestion
      const meeting = {...this.datasource[ID]};
```

I know we fixed it in the hook, but I think it's good to show "proper" code

Reply:
Actually, if we return a new object in `getMeeting` all the time, it should work there. Left a comment around that code instead